### PR TITLE
feat: [FFM-7527]: add click event to Environment cards

### PR DESCRIPTION
**Pull request by Kira-Jennings**
### Summary
- clicking on an Environment card redirects to Feature Flag details page with that card's environment selected

#### Screenshots
image

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by laura-mannus — DISMISSED** [review]

**Review by laura-mannus — APPROVED** [review]

File: src/modules/75-cf/hooks/useEnvironmentSelectV2.tsx [hunk 1]
```diff
@@ -73,10 +73,10 @@ export const useEnvironmentSelectV2 = (params: UseEnvironmentSelectV2Params) =>
...
   const {
-    loading: loadingAllEnvironmentsFlags,
-    error: fetchAllEnvironmentsFlagsError,
-    data: allEnvironmentsFlags,
-    refetch: refetchAllEnvironmentsFlags
+    loading: loadingProjectFlags,
+    error: fetchProjectFlagsError,
+    data: projectFlags,
+    refetch: refetchProjectFlags
   } = useGetProjectFlags({
```
Comment (crystal-fong) [inline src/modules/75-cf/hooks/useEnvironmentSelectV2.tsx:80]: was meant to ask in prev PR, why did we change the endpoint used here?

**Review by crystal-fong — APPROVED** [review]

File: src/modules/75-cf/hooks/useEnvironmentSelectV2.tsx (hunk 1 above)
Comment (knagurski) [reply src/modules/75-cf/hooks/useEnvironmentSelectV2.tsx:80]: @crystal-fong it's actually the same endpoint, it's just a different hook to use it.

Comment (Kira-Jennings) [reply src/modules/75-cf/hooks/useEnvironmentSelectV2.tsx:80]: I'm an idiot and had hardcoded it

File: src/modules/75-cf/pages/feature-flags/__tests__/FeatureFlagsPage.test.tsx [hunk 2]
```diff
@@ -220,6 +230,52 @@ describe('FeatureFlagsPage', () => {
...
+    userEvent.click(screen.getByText('common.allEnvironments'))
+
+    expect(refetchProjectFlags).toHaveBeenCalledTimes(1)
+    expect(screen.getAllByText('cf.environments.nonProd')).toHaveLength(17)
+    expect(screen.getAllByText('cf.environments.prod')).toHaveLength(17)
+
+    const rows = screen.getAllByRole('row')
+
+    // Feature Flag 1st row
+    const flag1Columns = getAllByRole(rows[1], 'cell')
```
Comment (knagurski) [inline src/modules/75-cf/pages/feature-flags/__tests__/FeatureFlagsPage.test.tsx:265]: Nothing wrong here, but just wanted to let you know that `within` exists and is great for this kinda thing. Both do the same thing in this case, so no need to change, just an FYI
```suggestion
    const flag1Columns = within(rows[1]).getAllByRole('cell')
```

Reads better IMO

**Review by knagurski — APPROVED** [review]
Nice 🔥

**Review by laura-mannus — APPROVED** [review]

**Review by knagurski — APPROVED** [review]
🔥

**Review by laura-mannus — APPROVED** [review]

**Pull request merged**